sim_3d_volumes.py

```diff
@@ -0,0 +1,127 @@
+"""Generate 3D map of molecules."""
+import os
+import numpy as np
+import torch
+import coords
+
+
+CUDA = torch.cuda.is_available()
+
+if CUDA:
+    Main_dir = "/scratch/ex-kdd-1/NicolasLegendre/Cryo/Data"
+else:
+    Main_dir = os.getcwd() + "\\Data\\"
+
+Particules = np.asarray([[1, 0, 0, 3], [0, 8, 0, 5], [0, 0, 7, 9]])
+N_particules = 2000
+Vol_size = 64
+Name = '4_points_3D'
+Long = 2
+
+
+def modify_weight(points, volume, size, long):
+    """
+    Fill an empty volume with particles.
+
+    Parameters
+    ----------
+    points : array, list particules position.
+    volume : ndarray, volume of the molecule.
+    size : int, size of the volume.
+    long : int, center the molecule around zero.
+
+    Returns
+    -------
+    volume : ndarray, volume of the molecule.
+
+    """
+    for point in points.T:
+        for i in range(size):
+            for j in range(size):
+                for k in range(size):
+                    volume[i][j][k] += np.exp(-np.linalg.norm(
+                        [i/long-size/long/2, j/long - size/long/2,
+                         k/long-size/long/2]-point)**2/2)
+    return volume
+
+
+def simulate_volumes(particules, n_volumes, img_size, long=2):
+    """
+    Update a volume with new particles.
+
+    Parameters
+    ----------
+    particules : ndarray, volume of the molecule.
```

Comment:
depending on the functions, `particules` seems to refer either to a volume (here) or to positions of particles (below): use two different names?